src/actions/admin.js

```diff
@@ -547,3 +547,23 @@ export function disablePlugin(pluginId: string): ActionFunc {
         return {data: true};
     };
 }
+
+export function getOAuthAppInfo(clientId: String): ActionFunc {
+    return bindClientFunc({
+        clientFunc: Client4.getOAuthAppInfo,
+        params: [clientId],
```

Comment:
is there a reducer that goes along side this? cause I'm not seeing the data sent to any reducer here

Reply:
It only calls the `Client4.getOAuthAppInfo`, it works in this way in a webapp, I did not make any changes in the logic

Reply:
What would be your thoughts on storing the data returned returned by these endpoints in the redux state instead of in the component state as you're currently doing it [here](https://github.com/mattermost/mattermost-webapp/pull/2387/files#diff-9192ed86bdc9e05ec21d324bbf848198R100)? That'd require adding a few actions and a reducer, but they should be fairly simple ones.

Generally, we try to keep as much state as possible in the redux store, and I think it might be a good idea to do that here as well, at least for the list of authorized OAuth apps

Reply:
I think single source of truth is a good idea, components would be realy pure and following one pattern through out the code is good. But there is another small thing, when data is used very rarely do we need to keep it in a store. Both of them are good for this use case and I think I will go with a redux store as you
> try to keep as much state as possible in the redux store

Reply:
Thanks! If it ends up being too much work to do that, feel free to leave it in the view state. While I like the idea of moving everything into redux, we've definitely been burned a few times by the "everything" part of decisions like that before. You're definitely right about the rarely used data not really being fit for this store since it can be kept for a long time when we only really need it while that one modal is open.

Reply:
Yes, I think keeping it simple in this case is a good fit